org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/device/DetectorView.java

```diff
@@ -332,12 +342,58 @@ protected void configure() {
 		try {
 			IRunnableDevice<Object> device = dservice.getRunnableDevice(info.getName());
 			Object model = info.getModel();
+
+			// Pass null to 'clear' the validation results view
+			selectionProvider.fireSelection(new StructuredSelection(new ValidateResults(info.getName(), null)));
+			
+			Object validateReturn = device.validateWithReturn(model);
+
+			ValidateResults validateResults = new ValidateResults(info.getName(), validateReturn);
+				
+			try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1141)